manage.py

```diff
@@ -0,0 +1,141 @@
+            if name in multi_value_properties:
+                code['properties'][name].add(value)
+            elif name in code['properties']:
+                raise Exception(f"{name} set to {code['properties'][name]}, not {value}")
+            else:
+                code['properties'][name] = value
+
+    not_selectable = {code['code'] for code in data['concept'] if code['properties'].get('notSelectable')}
+    logger.info('%s properties: %s', codelist, sorted(properties))
+
+    codes = []
+    for code in data['concept']:
+        if (
+            not code['properties'].get('notSelectable')
+            and code['properties']['status'] == 'active'
+            and any(parent in not_selectable for parent in code['properties']['subsumedBy'])
+        ):
+            codes.append(code)
+
+    return codes, not_selectable
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+def update_container():
+    """
+    Update schema/codelists/container.csv from HL7.
+    """
+    # https://terminology.hl7.org/CodeSystem/medicationknowledge-package-type/
+    data = requests.get('https://terminology.hl7.org/CodeSystem-medicationknowledge-package-type.json').json()
+
+    with csv_dump('container.csv', ['Code', 'Title']) as writer:
+        writer.writerows([[code['code'], code['display']] for code in data['concept']])
+
+
+@cli.command()
+def update_administration_route():
+    """
+    Update schema/codelists/administrationRoute.csv.
+    """
+    # https://terminology.hl7.org/CodeSystem/v3-RouteOfAdministration/
+    codes, not_selectable = hl7('RouteOfAdministration')
+
+    # "definition" is not used for Description, because it is the same as the "display", except for:
+    #
+    # - "Inhalation, respiratory", "Inhalation, oral"
+    # - "Injection, intrauterine", "Injection, intracervical (uterus)"
+    # - "instillation, urethral", "Instillation, urethral" (lettercase change)
+    # - "Topical application, vaginal", "Insertion, vaginal" (typographical error)
+
+    with csv_dump('administrationRoute.csv', ['Code', 'Title']) as writer:
+        for code in codes:
+            if code['properties']['synonymCode']:
+                # Prefer IPINHL to its synonyms.
+                if code['code'] in ('ORINHL', 'RESPINHL'):
+                    continue
+                elif code['code'] != 'IPINHL':
+                    logger.warning('RouteOfAdministration: unexpected synonymous code: %s', code)
+            writer.writerow([code['code'], code['display'][0].upper() + code['display'][1:]])
+
+
+@cli.command()
+def update_dosage_form():
+    """
+    Update schema/codelists/dosageForm.csv from HL7.
+    """
+    # https://terminology.hl7.org/CodeSystem/v3-orderableDrugForm/
+    codes, not_selectable = hl7('orderableDrugForm')
+
+    with csv_dump('dosageForm.csv', ['Code', 'Title', 'Description']) as writer:
+        for code in codes:
+            if 'SPRY' in code['code'] or code['code'] != 'SPRY':
```

Comment:
`or code['code'] != 'SPRY'` is this correct?

Reply:
Fixed in 35924ef

Reply:
so we only want the codes that contain 'SPRY'?